### PR TITLE
CHERI baremetal

### PR DESCRIPTION
**Pull request by marnovandermaas**
The most important change in this PR is enabling CHERI permissions checking in Ibex, adding capability tag bits to the TLUL user fields and compiling a sanity check program.

In more detail this PR:
- Vendors in the latest CHERIoT ibex with the changes necessary for to pass Verilator's lint.
- Patches upstream TLUL to add capability tags to the user field.
- Enable CHERI protection mode in Ibex.
- Add an SRAM block for capability tag bits.
- Add a CHERI baremetal sanity check program that blinks all the LEDs on the Sonata board (inverting based on inputs).

### CODE REVIEW

File: rtl/system/sonata_system.sv [hunk 1]
```diff
@@ -12,8 +12,8 @@
 // - Debug module.
 // - SPI for driving LCD screen.
 module sonata_system #(
-  parameter int unsigned GpiWidth = 8,
-  parameter int unsigned GpoWidth = 16,
+  parameter int unsigned GpiWidth = 13,
+  parameter int unsigned GpoWidth = 24,
```
Comment (HU90m) [inline rtl/system/sonata_system.sv:16]: Is this change necessary?

**Review by HU90m — APPROVED** [review]

File: rtl/system/sonata_system.sv (hunk 1 above)
Comment (marnovandermaas) [reply rtl/system/sonata_system.sv:16]: Not absolutely necessary, but they are better defaults for the Sonata board. The original numbers were based on the GPIO on the Arty A7 board.

Comment (HU90m) [reply rtl/system/sonata_system.sv:16]: Ah fair doozies

File: sw/cpp/cheri_sanity/.gitignore [hunk 2]
```diff
@@ -0,0 +1,8 @@
+boot.S.o
+boot.cc.o
+boot.elf
+boot.dump
+boot.bin
+cpu0_irom.vhx
+cpu0_irom.vmem
+end.S.o
```
[the remark this answers is not in the document]
Comment (marnovandermaas) [reply sw/cpp/cheri_sanity/.gitignore]: Thanks, that's definitely better. I've made this change now.
